src/app/pages/register/register.component.ts

```diff
@@ -0,0 +1,32 @@
+import { Component } from '@angular/core';
+import { AuthService } from '../../services/auth.service';
+import { Router } from '@angular/router';
+
+@Component({
+  templateUrl: './register.component.html',
+  styleUrls: ['./register.component.scss']
+})
+export class RegisterComponent {
+  registerFormData: {
+    username: string,
+    password: string,
+  } = {
+      username: '',
+      password: ''
+    };
+
+  constructor( 
+    private auth: AuthService,
+    private router: Router
+  ) { }
+
+  register() {
+    this.auth.register(this.registerFormData)
+    .then(()=> {
+      this.router.navigate(['/login']);
+    })
+    .catch((err) => { 
+      this.router.navigate(['/register']);
```

Comment:
Instead of redirecting them back to this page (the same page they are already on) it might be better to just inform the user of what went wrong.